main.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const ball = document.querySelector('.ball');
+const field = document.querySelector('.field');
+
+field.addEventListener('click', event => {
+  const ballWidth = ball.getBoundingClientRect().width;
+  const ballHeight = ball.getBoundingClientRect().height;
+  const borderWidth = Number(getComputedStyle(field).borderWidth.replace('px', ''));
+  const mousePosX = event.clientX;
+  const mousePosY = event.clientY;
+  const fieldWidth =field.getBoundingClientRect().width;
+  const fieldHeight = field.getBoundingClientRect().height; 
+
+
+ if (mousePosX < borderWidth + ballWidth / 2) {
+  ball.style.left = '0px';
+ } else  if (mousePosX > fieldWidth - borderWidth * 2 - ballWidth / 2) {
+  ball.style.left = `${fieldWidth - borderWidth * 2 - ballWidth}px`;
+ } else {
+  ball.style.left = `${mousePosX - ballWidth / 2 - borderWidth}px`;
```

Comment:
you do not calculate positions before assign as I wrote. each time you need to write template string and px.
 Create variable instead and assign it to style after `if`s